### ISSUE TITLE
Syntax Allen Brain interpolation, replace nan's with interpolated_spike_voltages

### ISSUE THREAD

**Issue by russelljjarvis** [scidash/neuronunit#180]
```
vm = self.results['voltage']
isv = self.results['interpolated_spike_voltage']
voltage = [ isv for i in vm[numpy.isnan(vm)] ]
```

**Comment by JustasB** [comment]
Could also do `parse_glif`, then convert the LEMS file to NEURON with jnml, and then check the output. NEURON does not support NaN values so PG had to decide how to deal with them when he converted the original models. You could also examine the NeuroML glif definition to see how he ensures all values are defined:

https://github.com/OpenSourceBrain/AllenInstituteNeuroML/blob/master/CellTypesDatabase/models/GLIF/GLIFs.xml

**Comment by russelljjarvis** [comment]
Thanks, Justas yes I will take the parse glif_approach for sure.
![rheobase](https://user-images.githubusercontent.com/7786645/39941239-c77ebce4-5510-11e8-997a-01f118a3832c.png)

The reason, the waveform looks like it pertains to, two unrealistic spikes instead of one is 
because I am using the glif cells own spike counting function, instead of the alternative of just using neuronunits own spike_train function instead.
 
``` python
        vms = self.get_membrane_potential()
        from neuronunit.capabilities.spike_functions import get_spike_train
        return get_spike_train(vms)
```

code for this is now at  
https://github.com/russelljjarvis/neuronunit/blob/glif/neuronunit/models/interfaces/glif.py#L62-#L65

Through trial and error I learned that the glif cells simply have different units (seconds, voltage) as opposed to (ms, mV). Using the approach above I have learned that I can get smooth membrane potential waveforms, indicative of time constants and capacitive resistance, however I also think that the default glif models, may lead to very unrealistic traces under the default parameters.

Running dir over the glif model indicates a lot of object attributes that are actually changeable model parameters. If I attach these to the genetic algorithm, I think I would start to get realistic outputs.

```
(Pdb) dir(self.glif)
['AScurrent_dynamics_method', 'AScurrent_reset_method', 'C', 'El', 'G', 'R_input', 'TYPE', '__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', 'append_threshold_components', 'asc_amp_array', 'asc_tau_array', 'coeffs', 'configure_library_method', 'configure_method', 'dt', 'dynamics', 'from_dict', 'init_AScurrents', 'init_threshold', 'init_voltage', 'k', 'reset', 'run', 'spike_cut_length', 'tau_m', 'th_adapt', 'th_inf', 'threshold_components', 'threshold_dynamics_method', 'threshold_reset_method', 'to_dict', 'type', 'voltage_dynamics_method', 'voltage_reset_method']
```

**Comment by russelljjarvis** [comment]
As discussed I have attached the GLIF cells to the BluePyOpt optimizer.

The result is more realistic waveforms, but still not realistic enough to do spike width and height tests.

![rheobase](https://user-images.githubusercontent.com/7786645/39950176-590a605a-5534-11e8-9ba1-776d74715d5c.png)

**Comment by rgerkin** [comment]
Right, the GLIF model is definitely not meant to capture anything about spike waveform properties, only spike timing and maybe a little bit of simple subthreshold behavior.

**Comment by pgleeson** [comment]
@russelljjarvis @rgerkin Glad to see you're using the GLIFs from the Allen Institute OSB repo. I haven't been following all of the discussions here, but if you do have any issue with running these, do open up issues in the https://github.com/OpenSourceBrain/AllenInstituteNeuroML repo.

BTW, as you'll probably have seen the cells are I&F (so no real spikes), but also the resting potentials of the cells are all shifted to 0mV to ease tuning/comparison between cells. This would probably mean many of the measures of whether they're like the equivalent bio cell can be expected to fail. The GLIFs could be adjusted to be more like the real ephys data, but that's a project in itself...
